server/command.go

```diff
@@ -0,0 +1,233 @@
+		action = split[1]
+	}
+	if len(split) > 2 {
+		parameters = split[2:]
+	}
+
+	if command != "/"+msteamsCommand {
+		return &model.CommandResponse{}, nil
+	}
+
+	if action == "link" {
+		return p.executeLinkCommand(c, args, parameters)
+	}
+
+	if action == "unlink" {
+		return p.executeUnlinkCommand(c, args)
+	}
+
+	if action == "show" {
+		return p.executeShowCommand(c, args)
+	}
+
+	return cmdError(args.ChannelId, "Unknown command. Valid options: link, unlink and show.")
+}
+
+func (p *Plugin) executeLinkCommand(c *plugin.Context, args *model.CommandArgs, parameters []string) (*model.CommandResponse, *model.AppError) {
+	if len(parameters) < 2 {
+		return cmdError(args.ChannelId, "Invalid link command, please pass the MS Teams team id and channel id as parameters.")
+	}
+
+	if !p.checkEnabledTeamByTeamId(args.TeamId) {
+		return cmdError(args.ChannelId, "This teams is not enabled for MS Teams sync.")
+	}
+
+	channel, appErr := p.API.GetChannel(args.ChannelId)
+	if appErr != nil {
+		return cmdError(args.ChannelId, "Unable to get the current channel information.")
+	}
+
+	canLinkChannel := channel.Type == model.ChannelTypeOpen && p.API.HasPermissionToChannel(args.UserId, args.ChannelId, model.PermissionManagePublicChannelProperties)
+	canLinkChannel = canLinkChannel || (channel.Type == model.ChannelTypePrivate && p.API.HasPermissionToChannel(args.UserId, args.ChannelId, model.PermissionManagePrivateChannelProperties))
+	if !canLinkChannel {
+		return cmdError(args.ChannelId, "Unable to link the channel, you has to be a channel admin to link it.")
+	}
+
+	channelsLinkedData, appErr := p.API.KVGet("channelsLinked")
+	if appErr != nil {
+		return cmdError(args.ChannelId, "Unable to get the linked channels information, please try again.")
+	}
+	channelsLinked := map[string]ChannelLink{}
+	json.Unmarshal(channelsLinkedData, &channelsLinked)
+	_, ok := channelsLinked[channel.TeamId+":"+channel.Id]
+	if ok {
+		return cmdError(args.ChannelId, "A link for this channel already exists, please remove unlink the channel before you link a new one")
+	}
+
+	_, err := p.msteamsAppClient.GetChannel(parameters[0], parameters[1])
+	if err != nil {
+		return cmdError(args.ChannelId, "MS Teams channel not found.")
+	}
+
+	channelsLinked[channel.TeamId+":"+channel.Id] = ChannelLink{
+		MattermostTeam:    channel.TeamId,
+		MattermostChannel: channel.Id,
+		MSTeamsTeam:       parameters[0],
+		MSTeamsChannel:    parameters[1],
+	}
+
+	channelsLinkedData, err = json.Marshal(channelsLinked)
+	if err != nil {
+		return cmdError(args.ChannelId, "Unable to store the new link, please try again.")
+	}
+
+	appErr = p.API.KVSet("channelsLinked", channelsLinkedData)
+	if appErr != nil {
+		return cmdError(args.ChannelId, "Unable to store the new link, please try again.")
+	}
+
+	p.restart()
+
+	return &model.CommandResponse{
+		ResponseType: model.CommandResponseTypeEphemeral,
+		Text:         "The msteams channel is now linked to this mattermost channel",
```

Comment:
0/5, maybe even set it to `Microsoft Teams`

```suggestion
		Text:         "The MS Teams channel is now linked to this mattermost channel",
```

Reply:
I think we can decide it later, for now lets keep "MS Teams" for consistency everywhere, and we can change it later.